### PR TITLE
Added data resiliency card

### PR DESCRIPTION
**Pull request by afreen23**
-fixtures ans tests for it
-component
[kubevirt/web-ui/pull/276](https://github.com/kubevirt/web-ui/pull/276)

build
dataisresilient
nodata

### CODE REVIEW

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '../../Dashboard/DashboardCard';
+import { StorageOverviewContext } from '../StorageOverviewContext';
+import { InlineLoading } from '../../Loading';
+import { getDataResiliencyData } from '../../../selectors/prometheus/selectors';
+
+const DataResiliencyStatusBody = ({ status }) =>
+  status ? (
+    <React.Fragment>
```
Comment (mareklibra) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: nitpick: the rest of web-ui-components code has `Fragment` in import explicitly.

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      <div className="kubevirt-data-resiliency__status-title--ok">Your Data Is Resilient</div>
+      <div className="kubevirt-data-resiliency__icon--ok">
+        <Icon type="fa" name="check-circle" size="3x" />
+      </div>
+    </React.Fragment>
+  ) : (
+    <React.Fragment>
+      <div className="kubevirt-data-resiliency__icon--error">
```
Comment (mareklibra) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: Can we use hust a single dash (`-`)? Throughout the code.

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Data Resiliency</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody
+        className="kubevirt-data-resiliency__dashboard-body"
+        isLoading={!(activePgRaw && totalPgRaw)}
+        LoadingComponent={InlineLoading}
```
Comment (mareklibra) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: The reason for passing the `LoadingComponent` down is the ability to reuse two different implementations.
The `web-ui` recently uses non-patternfly implementation. For consistency, its use has priority.
Can you please pass the component down from the web-ui? Please have a look at `ClusterOverview` for how to.
The `InlindeLoading` is a default and so should be set within `defaultProps`.

FYI: once openshift/console addopts patternfly's component, we will get rid of this passing.

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 1 above)
Comment (rawagner) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: I would say that we use `React.Fragment` more frequently than `Fragment` import :)

File: sass/components/StorageOverview/data-resiliency.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.kubevirt-data-resiliency__title {
+  margin-bottom: 80px;
```
Comment (rawagner) [inline sass/components/StorageOverview/data-resiliency.scss]: please use `em` units everywhere

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    <React.Fragment>
+      <div className="kubevirt-data-resiliency__status-title--ok">Your Data Is Resilient</div>
+      <div className="kubevirt-data-resiliency__icon--ok">
+        <Icon type="fa" name="check-circle" size="3x" />
+      </div>
+    </React.Fragment>
+  ) : (
+    <React.Fragment>
+      <div className="kubevirt-data-resiliency__icon--error">
+        <Icon type="fa" name="exclamation-triangle" size="1x" />
```
Comment (suomiy) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: why do we use different size for the icons?

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import { StorageOverviewContext } from '../StorageOverviewContext';
+import { InlineLoading } from '../../Loading';
+import { getDataResiliencyData } from '../../../selectors/prometheus/selectors';
+
+const DataResiliencyStatusBody = ({ status }) =>
+  status ? (
+    <React.Fragment>
+      <div className="kubevirt-data-resiliency__status-title--ok">Your Data Is Resilient</div>
+      <div className="kubevirt-data-resiliency__icon--ok">
+        <Icon type="fa" name="check-circle" size="3x" />
```
Comment (suomiy) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: is it possible to set the className just on the icon instead?

File: src/selectors/prometheus/selectors.js [hunk 7]
```diff
@@ -28,3 +28,8 @@ export const getLastUtilizationStat = response => {
   const history = getUtilizationVectorStats(response);
   return history ? history[history.length - 1] : null;
 };
+
+export const getDataResiliencyData = response => {
```
Comment (suomiy) [inline src/selectors/prometheus/selectors.js]: this function is the same as `getCapacityStats`. Can you please rename it and use it instead?

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 5 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: fixed it!! ...

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 2 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: `kubevirt-data-resiliency__icon-error` like this ?

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 6 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: I tried but did not get desired changes!

File: src/selectors/prometheus/selectors.js (hunk 7 above)
Comment (afreen23) [reply src/selectors/prometheus/selectors.js]: done

File: sass/components/StorageOverview/data-resiliency.scss (hunk 4 above)
Comment (afreen23) [reply sass/components/StorageOverview/data-resiliency.scss]: done

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 3 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: done ..thanx for the information

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 1 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: done !! yeah @rawagner frequently!!

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+      <div className="kubevirt-data-resiliency__icon-ok">
+        <Icon type="fa" name="check-circle" size="2x" />
+      </div>
+    </Fragment>
+  ) : (
+    <Fragment>
+      <div className="kubevirt-data-resiliency__icon-error">
+        <Icon type="fa" name="exclamation-triangle" size="2x" />
+      </div>
+      <div className="kubevirt-data-resiliency__status-title-error">Some Error Occurred</div>
```
Comment (suomiy) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: can we pass the error message here?

also  just `Error Occured` or just `Error` sounds better IMO

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  totalPgRaw: PropTypes.object,
+  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+};
+
+DataResiliencyBuildBody.propTypes = {
+  progressPercentage: PropTypes.number,
+};
+
+DataResiliencyStatusBody.propTypes = {
+  status: PropTypes.bool,
```
Comment (suomiy) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: please use something like `isStatusSuccessful` or `isSuccess` if we do not pass a status object.

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 8 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: what error message @suomiy

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 9 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: oh right! I am bad with names..thanx!!

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 8 above)
Comment (suomiy) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: that is what the user would say :) when the UI says `Some Error Occurred`

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  cleanPgRaw: PropTypes.object,
+  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+};
+
+DataResiliencyBuildBody.propTypes = {
+  progressPercentage: PropTypes.number,
+};
+
+DataResiliencyStatusBody.propTypes = {
+  status: PropTypes.bool,
```
Comment (suomiy) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: still missing here

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 8 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: what should I write ..can you tell ?

Comment (rawagner) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: Prometheus just returns integer value which indicates state of storage. So it seems like there is no way to get some meaningful error message right now. This is not blocking IMO and we can merge once the PR is rebased

Comment (suomiy) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: okay

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 10 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: done

File: src/components/StorageOverview/DataResiliency/DataResiliency.js (hunk 2 above)
Comment (afreen23) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: @mareklibra

**Comment by afreen23** [comment]
@mareklibra @suomiy @rawagner please review the PR..I have made requested changes

File: src/components/StorageOverview/DataResiliency/DataResiliency.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+  );
+};
+
+DataResiliency.defaultProps = {
+  totalPgRaw: null,
+  cleanAndActivePgRaw: null,
+  LoadingComponent: InlineLoading,
+};
+
+DataResiliencyStatusBody.defaultProps = {
```
Comment (suomiy) [inline src/components/StorageOverview/DataResiliency/DataResiliency.js]: please put the props next to the component. I think it is easier to read.

Comment (suomiy) [reply src/components/StorageOverview/DataResiliency/DataResiliency.js]: k, mergerd, nvm

**Pull request merged**